egs/yoloxochitl_mixtec/asr1/run.sh

```diff
@@ -0,0 +1,251 @@
+
+. ./path.sh || exit 1;
+. ./cmd.sh || exit 1;
+
+# general configuration
+backend=pytorch
+stage=0        # start from 0 if you need to start from data preparation
+stop_stage=100
+ngpu=1         # number of gpus ("0" uses cpu, otherwise use gpu)
+debugmode=1
+N=0            # number of minibatches to be used (mainly for debugging). "0" uses all minibatches.
+verbose=0      # verbose option
+resume=        # Resume the training from snapshot
+
+# dataset related
+annotation_type=eaf
+annotation_id=mixtec_surface
+text_format=surface # underlying_full, underlying_reduce
+
+# wav and transcription data directoy
+download_dir=
+wavdir=${download_dir}/Sound-files-Narratives-for-ASR
+annodir=${download_dir}/Transcriptions-for-ASR/ELAN-files-with-underlying-and-surface-tiers
+
+# feature configuration
+do_delta=false
+
+train_config=conf/train.yaml
+lm_config=conf/lm.yaml
+decode_config=conf/decode.yaml
+
+# rnnlm related
+lm_resume=        # specify a snapshot file to resume LM training
+lmtag=${annotation_id}        # tag for managing LMs
+
+# decoding parameter
+recog_model=model.acc.best # set a model to be used for decoding: 'model.acc.best' or 'model.loss.best'
+n_average=10
+
+dumpdir=dump   # directory to dump full features
+
+# bpemode (unigram or bpe)
+nbpe=150
+bpemode=unigram
+
+# exp tag
+tag=${annotation_id} # tag for managing experiments.
+
+. utils/parse_options.sh || exit 1;
+
+# Set bash to 'debug' mode, it will exit on :
+# -e 'error', -u 'undefined variable', -o ... 'error in pipeline', -x 'print commands',
+set -e
+set -u
+set -o pipefail
+
+train_set=train_${annotation_id}
+train_dev=dev_${annotation_id}
+test_set=test_${annotation_id}
+recog_set="${train_set} ${train_dev} ${test_set}"
+
+if [ ${stage} -le -1 ] && [ ${stop_stage} -ge -1 ]; then
+    local/download_and_untar.sh ${download_dir} http://www.openslr.org/resources/89/Yoloxochitl-Mixtec-Data.tgz Yoloxochitl-Mixtec-Data.tgz
+fi
+
+if [ ${stage} -le 0 ] && [ ${stop_stage} -ge 0 ]; then
+    ### Task dependent. You have to make data the following preparation part by yourself.
+    ### Task dependent. You have to make data the following preparation part by yourself.
+    python local/data_prep.py -w $wavdir -a $annodir -t data/${annotation_id} \
+                              -m ${annotation_type} -i local/speaker_wav_mapping_mixtec_remove_reserve.csv \
+                              -f ${text_format}
+
+    data/${annotation_id}/remix_script.sh
+
+    # Kaldi Version Split
+    # ./utils/subset_data_dir_tr_cv.sh data/${annotation_id} data/${train_set} data/recog
+    # ./utils/subset_data_dir_tr_cv.sh --cv-spk-percent 50 data/recog data/${train_dev} data/${test_set}
+
+    # ESPNet Version (same as voxforge)
+    # consider duplicated sentences (does not consider speaker split)
+    # filter out the same sentences (also same text) of test&dev set from validated set
+    local/split_tr_dt_et.sh data/${annotation_id} data/${train_set} data/${train_dev} data/${test_set}
```

Comment:
Is it OK?
If we don't have some official split, this way is fine.

Reply:
> LGTM!
> I have some minor comments.
> 
> * Do you think we should maintain Mixtec-consultant-database-unicode_2019-12-25.txt and speaker_wav_mapping_mixtec_remove_reserve.csv under the repository? One option is to include them in the openslr data. Either is fine, but I'm bit hesitating to maintain such data under the espnet repository.

Sounds good to add them into the openslr. I will contact dan to add the database code there and move them from the espnet repo

Reply:
We currently do not have an official split. Do you think we should add one to the dataset? I think it's up to us

Reply:
Maybe we can define it by ourselves by making a list of train/dev/eval splits?
The above script only considers the transcript overlap that happened in Voxforge and old commonvoice. But probably, we could have a good speaker-level and transcript-level open set.

Reply:
Currently, there is 34 spk with 529 recordings (each corresponds to around 120 segmental utterances). In addition, I checked the overlapping rate for text. It seems that overlapping is not a problem if we divide them by recordings. The overlapping rate (the exact same utt number / total utt number) between recordings is around 0.5-1.5%.

I currently split them as:

set|spk | recordings
--|-- | --
train| 21 | 497
dev| 6 | 16
test |7 | 16

Any suggestions or comments?

Reply:
I think this looks fine for me.
Let's try to standardize it.
Probably, we can make a webpage to briefly describe the database information, link for opneslr, and the official decision of the train/dev/test?